File: Smart-Bus/Smart-Bus/Classes/Main/CentralControl/Schedule/View/ScheduleView/Shade/ShadeCell/SHSchduleShadeCell.m
//
//  SHSchduleShadeCell.m
//  Smart-Bus
//
//  Created by Mark Liu on 2017/12/5.
//  Copyright © 2017年 SmartHome. All rights reserved.
//

#import "SHSchduleShadeCell.h"
#import "SHMoodShadeStatusButton.h"

@interface SHSchduleShadeCell ()


/// 窗帘名称
@property (weak, nonatomic) IBOutlet UILabel *nameLabel;

/// 设置按钮
@property (weak, nonatomic) IBOutlet SHMoodShadeStatusButton *homeButton;

@end

@implementation SHSchduleShadeCell

/// 设置模型
- (void)setShade:(SHShade *)shade {

    _shade = shade;
    
    self.nameLabel.text = shade.shadeName;
    
    switch (shade.currentStatus) {
         
            // 忽略
        case SHShadeStatusUnKnow: {
         
            [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_BYPASS"] forState:UIControlStateNormal];
        }
            break;
            
            // 打开
        case SHShadeStatusOpen: {
            
            [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_OPEN"] forState:UIControlStateNormal];
        }
            break;
            
            // 关闭
        case SHShadeStatusClose: {
      
            [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_CLOSE"] forState:UIControlStateNormal];
        }
            break;
            
        default:
            break;
    }
}

/// 设置按钮点击
- (IBAction)homeButtonClick {
    
    
    TYCustomAlertView *alertView = [TYCustomAlertView alertViewWithTitle:nil message:nil isCustom:YES];
    
    // 打开
    [alertView addAction: [TYAlertAction actionWithTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_OPEN"] style:TYAlertActionStyleDefault handler:^(TYAlertAction *action) {
        
        [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_OPEN"] forState:UIControlStateNormal];
        
        if ([self.delegate respondsToSelector:@selector(editShade:currentStatus:)]) {
            
            [self.delegate editShade:self.shade currentStatus:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_OPEN"]];
        }
    }]];
    
    // 关闭
    [alertView addAction: [TYAlertAction actionWithTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_CLOSE"] style:TYAlertActionStyleDefault handler:^(TYAlertAction *action) {
        
        [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_CLOSE"] forState:UIControlStateNormal];
        
        if ([self.delegate respondsToSelector:@selector(editShade:currentStatus:)]) {
            
            [self.delegate editShade:self.shade currentStatus:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_CLOSE"]];
        }
        
    }]];
    
    // 忽略
    [alertView addAction: [TYAlertAction actionWithTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_BYPASS"] style:TYAlertActionStyleCancel handler:^(TYAlertAction *action) {
        
        [self.homeButton setTitle:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_BYPASS"] forState:UIControlStateNormal];
        
        if ([self.delegate respondsToSelector:@selector(editShade:currentStatus:)]) {
            
            [self.delegate editShade:self.shade currentStatus:[[SHLanguageTools shareLanguageTools] getTextFromPlist:@"MOOD_IN_ZONE" withSubTitle:@"SHADE_BYPASS"]];
        }
        
    }]];
    
    TYAlertController *alertController = [TYAlertController alertControllerWithAlertView:alertView preferredStyle:TYAlertControllerStyleAlert transitionAnimation:TYAlertTransitionAnimationDropDown];
    
    alertController.backgoundTapDismissEnable = YES;
    
    [[UIApplication sharedApplication].keyWindow.rootViewController presentViewController:alertController animated:YES completion:nil];
}

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.selectionStyle = UITableViewCellSelectionStyleNone;
    self.backgroundColor = [UIColor clearColor];
    
    if ([UIDevice is_iPad]) {
        
        self.nameLabel.font = [UIView suitFontForPad];
        self.homeButton.titleLabel.font = [UIView suitFontForPad];
    }
}

/// cell的行高
+ (CGFloat)rowHeightForShadeCell {
    
    if ([UIDevice is_iPad]) {
        
        return navigationBarHeight + statusBarHeight;
        
    } else if ([UIDevice is4_0inch] || [UIDevice is3_5inch]) {
        
        return tabBarHeight;
        
    }
    
    return navigationBarHeight;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: Smart-Bus/Smart-Bus/Classes/Main/CentralControl/Schedule/View/ScheduleView/Shade/ShadeCell/SHSchduleShadeCell.swift
//
//  SHSchduleShadeCell.swift
//  Smart-Bus
//
//  Created by Mark Liu on 2017/12/5.
//  Copyright © 2017年 SmartHome. All rights reserved.
//

import UIKit

/// 窗帘代理
protocol SHEditRecordShadeStatusDelegate {
    
    func edit(shade: SHShade, status: String)
}

class SHSchduleShadeCell: UITableViewCell {
    
    /// 窗帘
    var shade: SHShade? {
        
        didSet {
            
            guard let curtain = shade else {
                return
            }
            
            nameLabel.text = curtain.shadeName
            
            switch curtain.currentStatus {
                
            case .unKnow:
                homeButton.setTitle(
                    SHLanguageText.shadeIgnore,
                    for: .normal
                )
                
            case .open:
                homeButton.setTitle(
                    SHLanguageText.shadeOpen,
                    for: .normal
                )
                
            case .close:
                homeButton.setTitle(
                    SHLanguageText.shadeClose,
                    for: .normal
                )
                
            default:
                break
            }
        }
    }
    
    
    /// 代理
    var delegate: SHEditRecordShadeStatusDelegate?
    
    /// 行高
    static var rowHeight: CGFloat {
        
        if UIDevice.is_iPad() {
            
            return navigationBarHeight + statusBarHeight
            
        } else if UIDevice.is4_0inch() || UIDevice.is3_5inch() {
            
            return tabBarHeight
        }
        
        return navigationBarHeight
    }
    
    /// 窗帘名称
    @IBOutlet weak var nameLabel: UILabel!
    
    /// 设置按钮
    @IBOutlet weak var homeButton: SHMoodShadeStatusButton!
    
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        selectionStyle = .none
        backgroundColor = .clear
        
        if UIDevice.is_iPad() {
            
            let font = UIView.suitFontForPad()
            
            nameLabel.font = font
            homeButton.titleLabel?.font = font
        }
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        
        // Configure the view for the selected state
    }
    
    
    /// 设置按钮点击
    @IBAction func homeButtonClick() {
        
        guard let curtain = shade else {
            return
        }
        
        let alertView =
            TYCustomAlertView(title: nil,
                              message: nil,
                              isCustom: true
        )
        
        // 打开
        let openAction =
            TYAlertAction(
                title: SHLanguageText.shadeOpen,
                style: .default) {
                    (action) in
                    
                    
                    self.homeButton.setTitle(
                        SHLanguageText.shadeOpen,
                        for: .normal
                    )
                    
                    self.delegate?.edit(
                        shade: curtain,
                        status: SHLanguageText.shadeOpen
                    )
                    
        }
        
        alertView?.add(openAction)
        
        // 关闭
        let closeAction =
            TYAlertAction(
                title: SHLanguageText.shadeClose,
                style: .default) {
                    (action) in
                    
                    
                    self.homeButton.setTitle(
                        SHLanguageText.shadeClose,
                        for: .normal
                    )
                    
                    self.delegate?.edit(
                        shade: curtain,
                        status: SHLanguageText.shadeClose
                    )
                    
        }
        
        alertView?.add(closeAction)
        
        // 忽略
        let ignoreAction =
            TYAlertAction(
                title: SHLanguageText.shadeIgnore,
                style: .default) {
                    (action) in
                    
                    
                    self.homeButton.setTitle(
                        SHLanguageText.shadeIgnore,
                        for: .normal
                    )
                    
                    self.delegate?.edit(
                        shade: curtain,
                        status: SHLanguageText.shadeIgnore
                    )
                    
        }
        
        alertView?.add(ignoreAction)
        
        let alertController =
            TYAlertController(
                alert: alertView,
                preferredStyle: .alert,
                transitionAnimation: .dropDown
        )
        
        alertController?.backgoundTapDismissEnable = true
        
    UIApplication.shared.keyWindow?.rootViewController?.present(
            alertController!,
            animated: true,
            completion: nil
        )
    }
}
